gc/src/trace.rs

```diff
@@ -1,5 +1,18 @@
+/// The Finalize trait. Can be specialized for a specific type to define
+/// finalization logic for that type.
+pub trait Finalize {
+    fn finalize(&self);
```

Comment:
Basically because the user isn't defining the `Trace` definition for their type manually, they are supposed to `#[derive(Trace)]`, which means that they don't have the ability to specify finalize manually. Even with specialization, you can't have two `impl Trace for T {...}` blocks, one of which would be generated by the `derive` plugin, and the other would be written by the user which wants to implement `Finalize`.

Basically, if we were to put it on Trace, and someone was to use the plugin to implement trace, then they couldn't also manually implement Finalize, as their implementation would conflict.

For example: https://is.gd/w112yQ